user-console/source/org/pentaho/mantle/client/ui/xul/MantleController.java

```diff
@@ -875,6 +876,15 @@ public void setContentEditSelected( boolean selected ) {
    * Notifies currently active Javascript callback of an edit event.
    */
   public void editContentClicked() {
+    // set delay for edit/view mode switching
+    setContentEditEnabled( false );
+    Timer loadOverlayTimer = new Timer() {
+      public void run() {
+        setContentEditEnabled( true );
+      }
+    };
+    loadOverlayTimer.schedule( 1000 );
+
```

Comment:
The lowest value at which stable 800ms. 
When the timer is running, the user sees a disabled icon (Edit/View mode toggle) - at the time the request is fulfilled as before. This protection against double click, and a very frequent switching modes